app.js

```diff
@@ -13,324 +13,287 @@ if (!supportsLocalStorage()) {
+  const attendingFilter = document.querySelector('#attending');
+  const notAttendingFilter = document.querySelector('#notAttending');
+  const noResponseFilter = document.querySelector('#noResponse');
+  let ul = document.getElementById('invitedList');
 
-  //Fetch and parse stored invitee names
-  function getInviteeLIS() {
-    let inviteeLIS = localStorage.getItem('inviteeLIS');
-    if (inviteeLIS) {
-      return JSON.parse(inviteeLIS);
-    } else {
-      return [];
-    }
-  }
+  //fullInviteList is to contain all invitees added to the list, and the current state of their object.
+  let fullInviteList = [];
+  //renderedInviteList should mirror fullInviteList until a filter is applied, at which point, re-map to include only those who match the filter
+  let renderedInviteList = [];
+  //The local storage copy of storedInviteList should be updated to reflect the fullInviteList every time a change is made to it.
+  let storedInviteList = [];
+  //display gets marked up version of renderedInviteList each time a change is made to it. See reloadDOM.
+  let display = '';
 
-  //Stringify and add new names to the stored names list
-  function saveInviteeLIS(string) {
-    let inviteeLIS = getInviteeLIS();
-    if (!string || inviteeLIS.indexOf(string) > -1) {
-      return false;
-    }
-    inviteeLIS.push(string);
-    localStorage.setItem('inviteeLIS', JSON.stringify(inviteeLIS));
-    return true;
-  }
+  //Rewrites the local storage to sync up with the passed in list. Preferably always pass in full list not rendered.
+  const refreshLocalStorageList = (List) => {
+    localStorage.removeItem('storedInviteList');
+    localStorage.setItem('storedInviteList', JSON.stringify(List));
+  };
 
-  //Find and delete a name from the stored names list, then resave the list with the amended contents
-  function removeInviteeLIS(inviteeName) {
-    const currentStoredNames = getInviteeLIS();
-    for (i = 0 ; i < currentStoredNames.length ; i++) {
-      if (inviteeName === currentStoredNames[i]) {
-        currentStoredNames.splice(i , 1);
-        localStorage.removeItem('inviteeLIS');
-        for (i = 0 ; i < currentStoredNames.length ; i++) {
-          saveInviteeLIS(currentStoredNames[i]);
-        }
-      }
+  //Fetch the stored list and populate storedInviteList with it
+  const getLocalStorageList = () => {
+    storedInviteList = localStorage.getItem('storedInviteList');
+    if (storedInviteList) {
+      storedInviteList = JSON.parse(storedInviteList);
+      return storedInviteList;
     }
-  }
+  };
 
-  //Create and append div, label and checkbox for new filters
-  function createFilterDiv(labelName, checkboxName, divName, labelText) {
-    labelName.textContent = labelText;
-    checkboxName.type = 'checkbox';
-    divName.appendChild(checkboxName);
-    divName.appendChild(labelName);
-    filterConDiv.appendChild(divName);
-  }
+  //formats and prints each object in the rendered list to the screen
+  const reloadDOM = () => {
```

Comment:
You're going to love using something like react or vue because this is done automatically with state or props are changed. Just wait!